### PR TITLE
minor tidyup of user deletions

### PR DESCRIPTION
**Pull request by chris18890**
minor tidyup of user deletions

### CODE REVIEW

File: crm/modules/amazon_payment/amazon_payment.inc.php [hunk 1]
```diff
@@ -200,7 +200,7 @@ function amazon_payment_contact_delete ($amazon_payment_contact) {
     $res = mysqli_query($db_connect, $sql);
     if (!$res) crm_error(mysqli_error($res));
     if (mysqli_affected_rows($db_connect) > 0) {
-        message_register("Amazon contact info deleted for: " . theme('contact_name', $esc_cid));
+        message_register("Deleted Amazon contact info for: " . theme('contact_name', $esc_cid));
     }
```
Comment (ramgarden) [inline crm/modules/amazon_payment/amazon_payment.inc.php]: So just a grammar fix?

File: crm/modules/contact/contact.inc.php [hunk 2]
```diff
@@ -219,19 +219,19 @@ function contact_save ($contact) {
  */
 function contact_delete ($cid) {
     global $db_connect;
-    $contact = crm_get_one('contact', array('cid'=>$cid));
+    $esc_cid = mysqli_real_escape_string($db_connect, $cid);
+    $contact = crm_get_one('contact', array('cid'=>$esc_cid));
     if (empty($contact)) {
-        error_register("No contact with cid $cid");
+        error_register("No contact with cid $esc_cid");
         return;
```
Comment (ramgarden) [inline crm/modules/contact/contact.inc.php]: Just adding SQL Injection protection?

File: crm/modules/contact/contact.inc.php [hunk 3]
```diff
@@ -219,19 +219,19 @@ function contact_save ($contact) {
...
     // Notify other modules the contact is being deleted
     $contact = module_invoke_api('contact', $contact, 'delete');
     // Remove the contact from the database
-    $esc_cid = mysqli_real_escape_string($db_connect, $cid);
     $sql = "DELETE FROM `contact` WHERE `cid`='$esc_cid'";
     $res = mysqli_query($db_connect, $sql);
     if (!$res) crm_error(mysqli_error($res));
-    message_register('Deleted contact: ' . theme('contact_name', $contact));
+    message_register('Deleted contact info for: ' . theme('contact_name', $esc_cid));
 }
```
Comment (ramgarden) [inline crm/modules/contact/contact.inc.php]: Did you swap out an array ($contact from crm_get_one method) for an integer ($esc_cid)?  Does the call to "theme" function still work in this case?

File: crm/modules/member/data.inc.php [hunk 4]
```diff
@@ -288,19 +288,14 @@ function member_save ($member) {
...
     $esc_cid = mysqli_real_escape_string($db_connect, $cid);
     $sql = "DELETE FROM `member` WHERE `cid`='$esc_cid'";
     $res = mysqli_query($db_connect, $sql);
     if (!$res) crm_error(mysqli_error($res));
     $sql = "DELETE FROM `membership` WHERE `cid`='$esc_cid'";
     $res = mysqli_query($db_connect, $sql);
     if (!$res) crm_error(mysqli_error($res));
-    message_register("Deleted membership info for: $name");
+    message_register("Deleted membership info for: " . theme('contact_name', $esc_cid));
 }
```
Comment (ramgarden) [inline crm/modules/member/data.inc.php]: Does theme('contact_name', idNumber) return the contact's Name?

File: crm/modules/amazon_payment/amazon_payment.inc.php (hunk 1 above)
Comment (chris18890) [reply crm/modules/amazon_payment/amazon_payment.inc.php]: Yep :)

File: crm/modules/contact/contact.inc.php (hunk 2 above)
Comment (chris18890) [reply crm/modules/contact/contact.inc.php]: Yep :)

File: crm/modules/member/data.inc.php (hunk 4 above)
Comment (chris18890) [reply crm/modules/member/data.inc.php]: Indeed, same as for the other modules (see amazon/user/paypal modules for examples

**Comment by chris18890** [comment]
willmerge at the end of the week if there are no objections @ramgarden @elplatt

**Pull request merged**